lib/edit-engine/addon/routes/application.js

```diff
@@ -0,0 +1,28 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class ApplicationRoute extends Route {
+  @service content;
+
+  model(params) {
+    let model;
+    if (params.dataset_id) {
+      model = this.content.fetchDataset(params.dataset_id);
+    }
+    if (params.document_id) {
+      model = this.content.fetchDocument(params.document_id)
+    }
+    return model;
+  }
```

Comment:
Defining the routes like this means they are not nested so we don't get the resolved model from the parent route (because the parent route hasn't resolved it). But we do get params.dataset_id or params.document_id which we can use to fetch it (and maybe we've got it cached at the service level?).